ddp-workspace/projects/ddp-sdk/src/lib/components/address/addressEmbedded.component.ts

```diff
@@ -518,48 +510,41 @@ export class AddressEmbeddedComponent implements OnDestroy, OnInit {
-            withLatestFrom(this.state$),
-            filter(([_, state]) => !!state.activityInstanceGuid),
-            tap(() => busyCounter$.next(1)),
-            concatMap(([_, state]) => this.addressService.deleteTempAddress(state.activityInstanceGuid)),
-            catchError(() => {
-                this.logger.logDebug(this.LOG_SOURCE, 'Temp delete failed. This is OK.');
-                return of(null);
-            }),
-            tap(() => busyCounter$.next(-1))
-        );
-
         const canSaveRealAddress = (address: Address | null) =>
             this.enoughDataToSave(address) && this.meetsActivityRequirements(address);
 
-        // "Real" as opposed to "Temp"
+        // "Real" as opposed to "Temp". Important we return the saved address as properties are added on server
         const saveRealAddressAction$ = this.saveTrigger$.pipe(
             withLatestFrom(this.formErrorMessages$, currentAddress$),
             filter(([_, formErrors, addressToSave]) => {
-                const canSave = canSaveRealAddress(addressToSave);
-                if (!canSave) {
-                    this.block.hasValidAddress = false;
-                }
-                return !formErrors?.length && canSave;
+                return !formErrors?.length && canSaveRealAddress(addressToSave);
             }),
             tap(() => busyCounter$.next(1)),
             concatMap(([_, formErrors, addressToSave]) => this.addressService.saveAddress(addressToSave, false)),
-            tap(() => busyCounter$.next(-1)),
             removeTempAddressOperator(),
+            tap(() => busyCounter$.next(-1)),
             share()
         );
 
+        const deleteTempAddressAction$ = saveRealAddressAction$.pipe(
+            withLatestFrom(this.state$),
+            filter(([_, state]) => !!state.activityInstanceGuid),
+            tap(() => busyCounter$.next(1)),
+            concatMap(([_, state]) => this.addressService.deleteTempAddress(state.activityInstanceGuid)),
+            catchError(() => {
+                this.logger.logDebug(this.LOG_SOURCE, 'Temp delete failed. This is OK.');
+                return of(null);
+            }),
+            tap(() => busyCounter$.next(-1))
+        );
+
         /* If we get to show validations, we touch controls so we will show validations errors
             associate with controls
          */
         const touchFormOnSubmitWithBadAddressAction$ = this.validationRequested$.pipe(
             withLatestFrom(currentAddress$),
-            filter(([_, addressToSave]) => {
-                return !canSaveRealAddress(addressToSave)
-                    || (this.block.requireVerified && !this.block.hasValidAddress);
-            }),
-            tap(() => this.addressInputComponent.markAddressTouched())
+            filter(([_, addressToSave]) => !canSaveRealAddress(addressToSave)),
+            tap(() => this.addressInputComponent.touchAllControls())
```

Comment:
There is no method `touchAllControls`, it is changed/renamed in 11e4d8af187c509bb7a5863f54663906552a0fd4